### DISCUSSION TITLE
Better audio quality 3.6.2=>3.8.1 with auto small network buffers?

### DISCUSSION THREAD

**Discussion by ve3meo** [jamulussoftware/jamulus#2230] [General]
I just upgraded Jamulus server from 3.6.2 to 3.8.1 on Ubuntu 20.04LTS and was immediately struck by how much better it sounded on 3.8.1 clients on both Windows 10 and Raspberry Pi and the same audio adapter (ART USB MIX 4). With 3.6.2 on the server, there was so much dropout distortion with both auto jitter and small network buffers enabled that I would disable one or the other or both and push the static buffers up a notch or two from where they had settled on auto. Now, I'm happy to have them enabled for shorter latency as there is little sacrifice in quality.

My question is whether there is any reason why this improvement should be expected? I've searched through a year's worth of discussions and see nothing that suggests it. My Internet connection has been good for months with ping around 4-5ms and the small jitter buffers settle around 4-5. Overall delay 20-23ms. Weekly session participants last night thought it sounded better, too, but maybe they were just being polite after I said how much I struggled with the foreign language of Linux.

**Comment by gilgongo** [comment]
The only thing I can think of is perhaps an improvement to multi-threading? https://github.com/jamulussoftware/jamulus/pull/960 doesn't seem too likely though.

**Reply by ve3meo** [reply]
Mine is a single-core server normally in the lowest cost Availability mode and occasionally upgraded to Dedicated cpu. I've no inkling whether the changes in multithreading would have the effect I observed.

**Comment by ann0see** [comment]
QoS? Delay Panning? Probably we can’t find much else

**Reply by gilgongo** [reply]
Oh yes, QoS might I guess if you're lucky enough to have network/ISP connections that respect it?

**Reply by ve3meo** [reply]
There is no QoS feature apparent in my Bell HomeHub 3000 FttH modem/router. So, if it does not support QoS, then there would be no benefit if the Jamulus client and server do, correct?

**Reply by gilgongo** [reply]
Not sure about the details. I was under the impression that the feature was used by ISPs rather than domestic routers but I could be wrong.

**Reply by ve3meo** [reply]
QoS has been available on inexpensive modems and routers for years. It might appear disabled to the subscriber on the Bell box yet implemented in a static way since it handles television, telephone and Internet although I'm not sure if all of that is handled in one stream. Its speed test history is pretty consistently 1Gbps down and 0.8Gbps up with 3ms latency, without the option to block local network activity. I don't see that over Ethernet as my devices are 100Mbps.

**Reply by gilgongo** [reply]
> QoS has been available on inexpensive modems and routers for years. 

I think we may be talking about different things. [The feature in Jamulus](https://jamulus.io/wiki/Software-Manual#quality-of-service-qos) that was introduced in 3.8.0 means that Jamulus **itself** (client and/or server) implements a "service class" for traffic generated. My understanding is that this may be used by ISPs independently of whether QoS is enabled on your (domestic) router. But it's a bit vague, I admit.

**Comment by ve3meo** [comment]
I guess if nobody else has observed a perceptible improvement in the quality of sound (a different QoS!) with automatic small network buffers in the 3.8.1 client when a server has been upgraded from 3.6.2 to 3.8.1, then it's unlikely that a change in Jamulus did the trick. I'll just enjoy the benefit without knowing why it happened.

**Comment by pljones** [comment]
When did server-side packet re-ordering come in?  That did have a big effect on sound quality (subject to network routing issues).

**Reply by gilgongo** [reply]
3.6.0 it looks like: https://github.com/jamulussoftware/jamulus/releases/tag/r3_6_0
